sdk/program/src/vote/state/vote_state_versions.rs

```diff
@@ -1,14 +1,44 @@
-use super::{vote_state_0_23_5::VoteState0_23_5, *};
+use super::{vote_state_0_23_5::VoteState0_23_5, vote_state_1_10_40::VoteState1_10_40, *};
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub enum VoteStateVersions {
     V0_23_5(Box<VoteState0_23_5>),
+    V1_10_40(Box<VoteState1_10_40>),
     Current(Box<VoteState>),
 }
 
 impl VoteStateVersions {
-    pub fn new_current(vote_state: VoteState) -> Self {
-        Self::Current(Box::new(vote_state))
+    pub fn new(vote_state: VoteState, use_current: bool) -> Self {
```

Comment:
prefer to plumb the feature set down here instead of a specific use bool so we can just retain the plumbing for future use

also avoid the rename.  this is just feature gating the concept of "current"

Reply:
Yes I agree, again Rust defeated me, I could not figure out how to import the FeatureSet type into this file.  I guess I can keep trying.

Reply:
feel free to ask about the specific error on discord. we can figure it out